WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingView.swift

```diff
@@ -5,53 +5,69 @@ struct StoreOnboardingView: View {
     private let viewModel: StoreOnboardingViewModel
     private let taskTapped: (StoreOnboardingTask) -> Void
     private let viewAllTapped: (() -> Void)?
+    private let dismissAction: (() -> Void)?
 
     init(viewModel: StoreOnboardingViewModel,
          taskTapped: @escaping (StoreOnboardingTask) -> Void,
-         viewAllTapped: (() -> Void)? = nil) {
+         viewAllTapped: (() -> Void)? = nil,
+         dismissAction: (() -> Void)? = nil) {
         self.viewModel = viewModel
         self.taskTapped = taskTapped
         self.viewAllTapped = viewAllTapped
+        self.dismissAction = dismissAction
     }
 
     var body: some View {
-        VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: Layout.verticalSpacing) {
-            // Progress view
-            StoreSetupProgressView(isExpanded: viewModel.isExpanded,
-                                   totalNumberOfTasks: viewModel.taskViewModels.count,
-                                   numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
-
-            // Task list
-            VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
-                ForEach(Array(viewModel.taskViewModels.enumerated()), id: \.offset) { index, taskViewModel in
-                    let isLastTask = index == viewModel.taskViewModels.count - 1
-
-                    StoreOnboardingTaskView(viewModel: taskViewModel,
-                                            showDivider: !isLastTask) { task in
-                        taskTapped(task)
+        VStack {
+            Color(.systemColor(.systemGray6))
+                .frame(height: 16)
+                .renderedIf(!viewModel.isExpanded)
+
+            let verticalSpacing = viewModel.isExpanded ? Layout.VerticalSpacing.expandedMode : Layout.VerticalSpacing.collapsedMode
+            VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: verticalSpacing) {
+                DismissButton(action: dismissAction)
+                    .renderedIf(viewModel.isExpanded)
```

Comment:
💡 We should show dismiss button on navigation bar to make sure it is not be scrolled out of view when the view is scrolled up.

Reply:
Done in f3fef5e655139d8a02972da2ef5f2f4b50d31e13